blockmanager.go

```diff
@@ -838,6 +838,127 @@ func (b *blockManager) getUncheckpointedCFHeaders(
 	return err
 }
 
+// checkpointedCFHeadersQuery holds all information necessary to perform and
+// handle a query for checkpointed filter headers.
+type checkpointedCFHeadersQuery struct {
+	blockMgr    *blockManager
+	msgs        []wire.Message
+	checkpoints []*chainhash.Hash
+	stopHashes  map[chainhash.Hash]uint32
+	headerChan  chan *wire.MsgCFHeaders
+}
+
+// requests creates the query.Requests for this CF headers query.
+func (c *checkpointedCFHeadersQuery) requests() []*query.Request {
+	var reqs []*query.Request
+	for _, m := range c.msgs {
+		reqs = append(reqs, &query.Request{
+			Req:        m,
+			HandleResp: c.handleResponse,
+		})
+	}
+	return reqs
+}
+
+// handleResponse is the internal response handler used for requests for this
+// CFHeaders query.
+func (c *checkpointedCFHeadersQuery) handleResponse(req, resp wire.Message,
```

Comment:
Hmm, I guess for interfaces, it doesn't matter if things are exported or not?

Reply:
?

Reply:
As in this seems to adhere to the interface, but using a private method to do so?